packages/fxa-settings/src/models/Session.ts

```diff
@@ -99,6 +100,9 @@ export class Session implements SessionData {
       query: GET_LOCAL_SIGNED_IN_STATUS,
       data: { isSignedIn: true },
     });
+
+    // This ensures the isSignedIn status is updated when this method is awaited
+    return Promise.resolve();
```

Comment:
I'm not sure I understand why this is needed. How is this different than returning nothing at all?

Reply:
I thought the same... let me double check this. Without this, I was seeing a redirect from `/settings` back to `signin` because isSignedIn was still false [here](https://github.com/mozilla/fxa/blob/11678ecec48985b6f306d186e3c0474a49e6d830/packages/fxa-settings/src/components/App/index.tsx#L204-L205).

(Also looks like I need to conditionally check the config there and change the `hardNavigateToContentServer` to `navigate` if signin is on)

Reply:
OK yeah, I'm not seeing this anymore even when I remove it. Valerie had pointed that redirect out to me when she did an initial run through and I was seeing the same thing and saw `isSignedIn` wasn't reliably being updated before `SettingsRoutes` was being rendered and I thought for some reason this fixed it, but maybe it was something else I've changed because I can't repro anymore. I'll revert this, if you can double check it's still working as expected (successfully taking you from `signin_token_code` to `/settings`) then we should be good.